### DISCUSSION TITLE
multiple endpoint

### DISCUSSION THREAD

**Discussion by Evolve6996** [vpnhood/VpnHood#68] [Q&A] [answered]
hello 
thanks for bringing this feature to VPNhood its quite handy 👍  but I don't know how this work I edited JSON file like this:
 
{
  "RestAccessServer": null,
  "FileAccessServer": {
    "TcpEndPoints": [ "0.0.0.0:443","[::]:443","0.0.0.0:80","[::]:80" ],
    "SslCertificatesPassword": null
  },
  "IsDiagnoseMode": false
}

-- I don't know how it works is vpnhood now choose port automatically between 443 and 80 ?

-- can I choose it manually to which port my vpnhood client should connect?

**Comment by trudyhood** [comment, marked as answer]
Hi, You can't choose its port with VpnHood UI. it makes UI so complicated for many users.

The server can listen to multiple endpoints at once. In your case, there are 3 endpoints (2 IPV4 and 1 IPv6)
You have plenty of options. First, you can include all endpoints in your AccessKey, which I don't recommend for one server. In this case, a random endpoint will be selected by VpnHood. IpV6 will not be selected if a user doesn't have IPv6 network.
```
dotnet VpnHoodServer.dll gen
````

You can also generate different AccessKey and specify an endpoint for it. Please note that you need set your public IP and make sure it will end up to your server endpoints, such as:
```
dotnet VpnHoodServer.dll gen -ep 100.10.20.30:443
```
for more options
```
dotnet VpnHoodServer.dll gen -?
```

Have any question or idea, just contact us. :)
